lib/parallel_tests/cli.rb

```diff
@@ -182,6 +190,23 @@ def parse_options!(argv)
       options
     end
 
+    def parse_additional_test_options(argv)
+      file_args_at = (argv.rindex("--") || -1) + 1
+      files = argv[file_args_at..-1]
+
+      test_opts_at = (argv.index("--") || -2) + 1
+      test_opts_at = 0 if test_opts_at == file_args_at
+
+      test_opts_end = file_args_at - 1
+
+      test_options = argv[test_opts_at...test_opts_end] || []
+
+      [
+        (test_options if test_options.any?),
+        (files if files.any?)
+      ]
+    end
```

Comment:
that looks pretty good / I like the tests :)

maybe 1 more level better: grab until the -- twice and return that -> less logic / if-else

```Ruby
# returns files and test-options
# ['a', '--', 'b'] -> [['b'], ['a']]
def parse_additional_test_options(argv)
  2.times.map do
    start = (argv.rindex('--') || -1) + 1
    parsed = argv.slice!(start..-1)
    argv.pop # discard --
    parsed if parsed.any?
  end
end
```

or use partition

```
def parse_additional_test_options(argv)
  start = argv.rindex('--')
  start ? argv.delete_at(start) : start = -1
  i = -1
  argv.partition { |arg| i +=1; (i+=1) >= start }
end
```